torch/distributed/rpc/backend_registry.py

```diff
@@ -28,26 +38,44 @@ def register_backend(backend_name, init_backend_handler):
     # Create a new enum type, `BackendType`, with extended members.
     existing_enum_dict = {member.name: member.value for member in BackendType}
     extended_enum_dict = dict(
-        {backend_name: BackendValue(init_backend_handler=init_backend_handler)},
+        {
+            backend_name: BackendValue(
+                construct_rpc_agent_options_handler=construct_rpc_agent_options_handler,
+                init_backend_handler=init_backend_handler,
+            )
+        },
         **existing_enum_dict
     )
     BackendType = enum.Enum(value="BackendType", names=extended_enum_dict)
     return BackendType[backend_name]
 
 
+def construct_rpc_agent_options(
+    backend, rpc_timeout=rpc_constants.DEFAULT_RPC_TIMEOUT, **kwargs
+):
+    if not isinstance(rpc_timeout, datetime.timedelta):
+        raise RuntimeError("`rpc_timeout` must be a `datetime.timedelta`.")
+
+    return backend.value.construct_rpc_agent_options_handler(rpc_timeout, **kwargs)
+
+
 def init_backend(backend, *args, **kwargs):
     return backend.value.init_backend_handler(*args, **kwargs)
 
 
+def process_group_construct_rpc_agent_options_handler(
```

Comment:
This is a very long name, does `process_group_agent_options_handler` work here?

Reply:
Hey, I would keep it as is. 

It's long. Since it's not a user-facing API, I think it's not a pain for users. 

I am making the naming conforming to a schema here. i.e. <backend_name> + <attribute in the BackendValue type> + "handler"

Reply:
Please prefix internal functions with an `_` as a hint to readers that it's internal.

Reply:
Updated.